src/protocols/multi_party_ecdsa/gg_2020/party_i.rs

```diff
@@ -138,27 +141,33 @@ pub struct SignatureRecid {
     pub recid: u8,
 }
 
-pub fn generate_h1_h2_N_tilde() -> (BigInt, BigInt, BigInt, BigInt) {
-    //note, should be safe primes:
+pub fn generate_h1_h2_N_tilde() -> (BigInt, BigInt, BigInt, BigInt, BigInt) {
+    // note, should be safe primes:
     // let (ek_tilde, dk_tilde) = Paillier::keypair_safe_primes().keys();;
     let (ek_tilde, dk_tilde) = Paillier::keypair().keys();
     let one = BigInt::one();
     let phi = (&dk_tilde.p - &one) * (&dk_tilde.q - &one);
-    let h1 = BigInt::sample_below(&phi);
-    let S = BigInt::from(2).pow(256 as u32);
-    let xhi = BigInt::sample_below(&S);
-    let h1_inv = BigInt::mod_inv(&h1, &ek_tilde.n).unwrap();
-    let h2 = BigInt::mod_pow(&h1_inv, &xhi, &ek_tilde.n);
+    let h1 = BigInt::sample_below(&ek_tilde.n);
```

Comment:
Why is h1 now between `0...n` when before it was `0...phi`?

Reply:
Not sure why it was '0..phi', I think it should be an element of 'Z*_n'. Note deleted line 150 where h1_inv is taken modulo `n`, and not modulo `phi`